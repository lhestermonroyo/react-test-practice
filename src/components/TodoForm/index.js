import React from 'react';
import { TextField, InputAdornment, IconButton } from '@material-ui/core';
import { Add } from '@material-ui/icons';

const TodoForm = (props) => {
  const { todoData, handleChange, handleSubmit } = props;
  return (
    <form onSubmit={handleSubmit}>
      <TextField
        onChange={handleChange}
        value={todoData}
        name="todoData"
        label="Add Todo"
        margin="normal"
        variant="outlined" 
        color="primary" 
        fullWidth
        InputProps={{
          endAdornment: (
            <InputAdornment position="end">
              <IconButton type="submit" edge="end">
                <Add/>
              </IconButton>
            </InputAdornment>
          ),
        }}
        />
    </form>
  )
}

export default TodoForm;