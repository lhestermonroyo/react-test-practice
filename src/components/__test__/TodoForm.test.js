import React from 'react';
import { shallow } from 'enzyme';
import TodoForm from '../TodoForm';

describe('TodoForm', () => {
  let component;
  let props;

  beforeEach(() => {
    props = { 
      todoData: '',
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
    };
    component = shallow(<TodoForm {...props} />);
  })

  it('matches its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correct structure', () => {
    expect(component.is('div')).toBe(true);
  })
});
