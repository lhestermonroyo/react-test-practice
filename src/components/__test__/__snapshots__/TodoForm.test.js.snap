// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm matches its snapshot 1`] = `ShallowWrapper {}`;
